streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -383,7 +403,84 @@ private static void verifySum(final Map<TopicPartition, List<ConsumerRecord<byte
                     sum += value;
                     currentSumPerKey.put(key, sum);
 
-                    if (!receivedKey.equals(key) || receivedValue != sum) {
+                    if (!receivedKey.equals(key) || receivedValue != sum.longValue()) {
+                        throw new RuntimeException("Result verification failed for " + receivedRecord + " expected <" + key + "," + value + "> but was <" + receivedKey + "," + receivedValue + ">");
+                    }
+                }
+            } catch (final NullPointerException e) {
```

Comment:
We should check that `inputRecords.hasNext` is false after the for loop, or any manners to make sure that pairing lists have the same size.